containers/ecr-viewer/src/app/services/ecrMetadataService.ts

```diff
@@ -96,36 +120,69 @@ export const evaluateEcrMetadata = (fhirBundle: Bundle): EcrMetadata => {
     }
   };
 
-  const fhirERSDWarnings = evaluateAll(
+  const fhirEICRProcessingStatus = evaluateValue(
+    fhirBundle,
+    fhirPathMappings.eICRProcessingStatus,
+  );
+  const fhirEICRProcessingStatusReasonObs = evaluateOne(
     fhirBundle,
-    fhirPathMappings.eRSDwarnings,
+    fhirPathMappings.eICRProcessingStatusReason,
   );
-  const eRSDTextList: ERSDWarning[] = [];
-
-  for (const warning of fhirERSDWarnings) {
-    if (warning.code === "RRVS34") {
-      eRSDTextList.push({
-        warning:
-          "Sending organization is using an malformed eRSD (RCTC) version",
-        versionUsed: "2020-06-23",
-        expectedVersion:
-          "Sending organization should be using one of the following: 2023-10-06, 1.2.2.0, 3.x.x.x.",
-        suggestedSolution:
-          "The trigger code version your organization is using could not be determined. The trigger codes may be out date. Please have your EHR administrators update the version format for complete eCR functioning.",
-      });
-    } else if (warning.code === "RRVS29") {
-      eRSDTextList.push({
-        warning:
-          "Sending organization is using an outdated eRSD (RCTC) version",
-        versionUsed: "2020-06-23",
-        expectedVersion:
-          "Sending organization should be using one of the following: 2023-10-06, 1.2.2.0, 3.x.x.x.",
-        suggestedSolution:
-          "The trigger code version your organization is using is out-of-date. Please have your EHR administration install the current version for complete eCR functioning.",
-      });
+
+  function geteRSDInfo(
+    processingStatus: string | undefined,
+    reasonObs: Observation | undefined,
+  ): ERSDInfo | undefined {
+    if (processingStatus === "RRVS19") {
+      return { success: true };
+    } else if (processingStatus && !reasonObs) {
+      return unknownWarning;
+    } else if (!processingStatus && !reasonObs) {
+      return undefined;
     }
+
+    const coding = reasonObs?.valueCodeableConcept?.coding?.[0];
+    const warningCode = coding?.code ?? "";
+    const warningName =
+      coding?.display || eicrProcessingReasonMap[warningCode] || "";
+
+    let versionUsed: string | undefined;
+    let versionExpected: string | undefined;
+
+    reasonObs?.component?.forEach((component) => {
+      const detailVal = component.valueString;
+      const detailCode = component.code?.coding?.[0]?.code;
+      const detailDisplay = component.code?.coding?.[0]?.display;
+
+      if (!detailCode) return;
+
+      if (detailCode === "RRVS33") {
+        versionExpected = detailVal;
+      } else {
+        versionUsed = detailDisplay
+          ? `${detailDisplay}: ${detailVal}`
+          : detailVal;
+      }
+    });
+    return warningName || versionUsed || versionExpected || warningCode
+      ? {
+          success: false,
+          eRSDWarning: {
+            warning: warningName ?? noData,
```

Comment:
why `??` here instead of `||` - same question for `suggestedSolution`

Reply:
also, should we default in the unknown warning text instead of `noData` on the warning code?